scripts/icu/63.1-min-static-data/script.sh

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env bash
+
+# Build ICU common package (libicuuc.a) with data file separate and with support for legacy conversion and break iteration turned off in order to minimize size
+
+MASON_NAME=icu
+MASON_VERSION=63.1-min-static-data
+MASON_LIB_FILE=lib/libicuuc.a
+#MASON_PKGCONFIG_FILE=lib/pkgconfig/icu-uc.pc
+
+. ${MASON_DIR}/mason.sh
+
+MASON_BUILD_DEBUG=0 # Enable to build library with debug symbols
+MASON_CROSS_BUILD=0
+
+function mason_load_source {
+    mason_download \
+        http://download.icu-project.org/files/icu4c/63.1/icu4c-63_1-src.tgz \
+        3e24e6839878321f5193a3c4aa6f02210052fb31 \
+        source
+
+    mason_extract_tar_gz
+
+    mason_download \
+        http://download.icu-project.org/files/icu4c/63.1/icu4c-63_1-data.zip \
+        a0eb02a525822ea5ab409e04cf280bacccbb5d4d \
+        data
+
+    mason_extract_zip -d ${MASON_NAME}/source
+
+    export MASON_BUILD_PATH=${MASON_ROOT}/.build/${MASON_NAME}
+}
+
+function mason_prepare_compile {
+    if [[ ${MASON_PLATFORM} == 'ios' || ${MASON_PLATFORM} == 'android' || ${MASON_PLATFORM_VERSION} != `uname -m` ]]; then
+        mason_substep "Cross-compiling ICU. Starting with host build of ICU to generate tools."
+
+        pushd ${MASON_ROOT}/..
+        env -i HOME="$HOME" PATH="$PATH" USER="$USER" ${MASON_DIR}/mason build ${MASON_NAME} ${MASON_VERSION}
+        popd
+
+        # TODO: Copies a bunch of files to a kind of orphaned place, do we need to do something to clean up after the build?
+        #  Copying the whole build directory is the easiest way to do a cross build, but we could limit this to a small subset of files (icucross.mk, the tools directory, probably a few others...)
+        #  Also instead of using the regular build steps, we could use a dedicated built target that just builds the tools
+        mason_substep "Moving host ICU build directory to ${MASON_ROOT}/.build/icu-host"
+        rm -rf ${MASON_ROOT}/.build/icu-host
+        cp -R ${MASON_BUILD_PATH}/source ${MASON_ROOT}/.build/icu-host
+    fi
+}
+
+function trim_data {
```

Comment:
this function is what "patches" the ICU data files so that we only build some of the datasets and only for some of the locales. i could have used a patch file but ICU docs say to copy files and modify those so I regex'd my way there instead.

Reply:
👌